CONTRIBUTING.md

```diff
@@ -38,6 +38,29 @@ times in N different commits. If there was some accidental reformatting or white
 the PR.
 
+## API Design Guidelines
+
+### Capability Architecture
+
+The teams-js library provides a suite of APIs that encompass a broad range of functionality across multiple hosts. Different hosts will support different subsets of that functionality (e.g. Outlook may support different functionality than Teams).
+
+The concept of a "capability" has been defined to organize APIs and provide a host-agnostic method for detecting supported functionality. All functionality in the SDK is grouped into these capabilities.
+
+A capability is a logical grouping of APIs that provide similar functionality. A host supports a given capability _only if_ it supports all the APIs defined within that capability. Hosts **cannot** partially implement a capability. Capabilities can be feature or content-based, such a `mail`, `calendar`, `chat`, `dialog`, `authentication`, etc., but there may also be capabilities for application types such as `pages`, or other potential groups not yet anticipated.
+
+In teams-js, APIs are defined as functions in a JavaScript namespace whose name matches their required capability. If an app is running in a host that supports the `calendar` capability, then the app can safely call APIs such as `calendar.openCalendarItem` (as well as other calendar-related APIs defined in the namespace). Meanwhile, if an app attempts to call an API that's not supported in that host, the API will throw an exception.
+
+There are two ways for an app to take a dependency on a given capability:
+
+1. The app will be able to declare the capability as required in its manifest. Hosts will only load apps if they support all the capabilities those apps require. The app will not be listed in the hosts's store if any of its required capabilities are unsupported.
+2. If the app doesn't declare a capability as required, then it needs to check for that capability at runtime by calling an `isSupported()` function on that capability and adjust its behavior as appropriate. This allows an app to enable optional UI and functionality in hosts that support it, while continuing to run (and appear in the store) for hosts that don't.
+
+### Compatiblity and capabilities under development
+
+Since hosts must support all functionality in a capability to declare it as "supported," this generally means that _new functions cannot be added to existing shipped capabilities_. If new functions were added to shipped capabilities, older hosts would not have support for the new function and consequently the "all or nothing" capability promise would be violated. Since developing new capabilties necessitates some amount of iteration and support rollout time, new capabilities (and their functions) still under development should be TSDoc tagged with the [@beta](https://tsdoc.org/pages/tags/beta/) tag. This ensures that potential consumers are aware that any and all functionality in that capability can change in the future and that they should not use it in production apps.
```

Comment:
Should we talk about the sub capabilities that host can choose not to support inside a capability? e.g. dialog.bot

Reply:
Thank you for the suggestion. This triggered me to expand this PR a bit to include more documentation that we previously had written elsewhere, including a section on subcapabilities. I've now pulled the design guidance out of this doc into a separate document in this PR (it was getting long in here) and that doc talks also about subcapabilities. I've also added a new doc with guidance on unit testing.